### DISCUSSION TITLE
关于cgo这块的事件获取的问题

### DISCUSSION THREAD

**Discussion by zcl1115** [harmonycloud/kindling#629] [agent collector]
想问一下，collector跟cgo之间获取事件的步骤，是不是collector先初始化1000的一个指针列表，然后定期获取事件，而cgo里面做的操作就是判断遍历这个指针列表（最大为998），然后如果指针地址为空，就申请kindling_event_t_for_go的内存地址，然后后续对这个指向的指向的内容做修改，这1000的列表相当于初始化后是固定的

**Comment by dxsup** [comment]
是的，批量获取事件，会一直复用这个列表，但是到了Go的部分，还是会申请内存保存事件，然后用于后续处理
